HW2/P4/driver.py

```diff
@@ -15,14 +15,23 @@
 from timer import Timer
 import threading
 
-def py_median_3x3(image, iterations=10, num_threads=1):
+def py_median_3x3(image, iterations=10, num_threads = 4):
     ''' repeatedly filter with a 3x3 median '''
     tmpA = image.copy()
     tmpB = np.empty_like(tmpA)
 
     for i in range(iterations):
-        filtering.median_3x3(tmpA, tmpB, 0, 1)
+        # filtering.median_3x3(tmpA, tmpB, 0, 1)
         # swap direction of filtering
+        thread_list = []
```

Comment:
How do you make sure that threads don't start their iterations too soon? I don't see any locks/events or anything.